### PR TITLE
[TIMOB-25985] iOS/Android: Add "attributedString" API to Ti.UI.Button

### PR DESCRIPTION
**Pull request by hansemannn**
**JIRA:** https://jira.appcelerator.org/browse/TIMOB-25985

### CODE REVIEW

File: android/modules/ui/src/java/ti/modules/titanium/ui/widget/TiUIButton.java [hunk 1]
```diff
@@ -188,4 +201,14 @@ public void propertyChanged(String key, Object oldValue, Object newValue, KrollP
...
+
+	private void setAttributedStringText(AttributedStringProxy attrString)
+	{
+		AppCompatButton btn = (AppCompatButton) getNativeView();
+		Bundle bundleText =
+			AttributedStringProxy.toSpannableInBundle(attrString, TiApplication.getAppCurrentActivity());
+		if (bundleText.containsKey(TiC.PROPERTY_ATTRIBUTED_STRING)) {
+			btn.setText((Spannable) bundleText.getCharSequence(TiC.PROPERTY_ATTRIBUTED_STRING));
+		}
+	}
```
Comment (jquick-axway) [inline android/modules/ui/src/java/ti/modules/titanium/ui/widget/TiUIButton.java]: I don't think you need to call `toSpannableInBundle()` do you? We don't need to for labels, which uses Android's `TextView` Java class... andr Android `Button` class inherits from `TextView` as well. Also, the `setText()` method accepts a `CharSequence` type that a `Spannable` type inherits from. So, no casting should be needed. How about something like this?

```java
AppCompatButton btn = (AppCompatButton) getNativeView();
if (btn != null) {
	CharSequence text = AttributedStringProxy.toSpannable(attrString, TiApplication.getAppCurrentActivity());
	if (text == null) {
		text = "";
	}
	btn.setText(text);
}
```

**Review by jquick-axway — CHANGES_REQUESTED** [review]

File: android/modules/ui/src/java/ti/modules/titanium/ui/widget/TiUIButton.java (hunk 1 above)
Comment (hansemannn) [reply android/modules/ui/src/java/ti/modules/titanium/ui/widget/TiUIButton.java]: It was basically copied from the label implementation, but I could change it!

Comment (jquick-axway) [reply android/modules/ui/src/java/ti/modules/titanium/ui/widget/TiUIButton.java]: Oh I see. You copied it from "TiUIText.java" which is our TextField/TextArea code (seems to be doing an unnecessary step). The way I'm doing it is based on the "TiUILabel.java" code.

Comment (hansemannn) [reply android/modules/ui/src/java/ti/modules/titanium/ui/widget/TiUIButton.java]: Perfect! Will change now.

**Review by jquick-axway — APPROVED** [review]
CR: Pass (Android)

**Review by vijaysingh-axway — APPROVED** [review]
CR approved for iOS.

**Pull request merged**